src/frontend/src/statement/ddl.rs

```diff
@@ -0,0 +1,570 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use api::helper::ColumnDataTypeWrapper;
+use api::v1::{column_def, AlterExpr, CreateTableExpr, TruncateTableExpr};
+use catalog::{CatalogManagerRef, DeregisterTableRequest, RegisterTableRequest};
+use chrono::DateTime;
+use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+use common_catalog::format_full_table_name;
+use common_meta::cache_invalidator::Context;
+use common_meta::ddl::ExecutorContext;
+use common_meta::ident::TableIdent;
+use common_meta::key::schema_name::{SchemaNameKey, SchemaNameValue};
+use common_meta::rpc::ddl::{DdlTask, SubmitDdlTaskRequest, SubmitDdlTaskResponse};
+use common_meta::rpc::router::{Partition, Partition as MetaPartition};
+use common_meta::table_name::TableName;
+use common_query::Output;
+use common_telemetry::info;
+use datatypes::prelude::ConcreteDataType;
+use datatypes::schema::RawSchema;
+use partition::partition::{PartitionBound, PartitionDef};
+use snafu::{ensure, OptionExt, ResultExt};
+use sql::ast::Value as SqlValue;
+use sql::statements::create::Partitions;
+use sql::statements::sql_value_to_value;
+use table::metadata::{self, RawTableInfo, RawTableMeta, TableId, TableInfo, TableType};
+use table::requests::{AlterTableRequest, TableOptions};
+use table::TableRef;
+
+use super::StatementExecutor;
+use crate::error::{
+    self, AlterExprToRequestSnafu, CatalogSnafu, ColumnDataTypeSnafu, ColumnNotFoundSnafu,
+    DeserializePartitionSnafu, ParseSqlSnafu, Result, SchemaNotFoundSnafu, TableAlreadyExistSnafu,
+    TableMetadataManagerSnafu, TableNotFoundSnafu, UnrecognizedTableOptionSnafu,
+};
+use crate::table::DistTable;
+use crate::MAX_VALUE;
+
+impl StatementExecutor {
+    pub fn catalog_manager(&self) -> CatalogManagerRef {
+        self.catalog_manager.clone()
+    }
+
+    pub async fn create_table(
+        &self,
+        create_table: &mut CreateTableExpr,
+        partitions: Option<Partitions>,
+    ) -> Result<TableRef> {
```

Comment:
It seems that it deals with gRPC DDL expressions rather than SQL statements, so it may not be appropriate to put these functions in `statement/ddl.rs`.